earth_enterprise/rpms/build.gradle

```diff
@@ -186,10 +158,8 @@ def rpmCommandFilter = {
     return rpmCapabilityMap.containsKey(it) ? rpmCapabilityMap[it] : it
 }
 
-
-
 ospackage {
-    version = openGeeVersion
+    version = "<version unset>"
```

Comment:
This will be set within a task if any of the package tasks need to be executed.   If none of the package tasks need to be executed then this doesn't need to be set.

Reply:
This will create issues you may not be expecting.  The value of the `version` property is used to calculate the name of output files for the tasks that generate RPMs.  If those values are miscalculated during the configuration phase the incremental building of these tasks will be broken.

For example, your repository is tagged as GEE version 5.2.3, you make a build of the RPMs, then you tag your repository as 5.2.4, and try to make another build.  The output files of the RPM build task for 5.2.4 may be the same as the ones for 5.2.3, because only the version changed, but, now that's not considered as part of the output, because the version is always set to `<version unset>` when calculating output files.  So, Gradle would tell you there's nothing to build, and you may be scratching your head wondering why it's not producing packages for 5.2.4.

Reply:
This is set by the `setGeeVersion` task and that task has not declared output so gradle will always run it even in incremental builds if it is called directly or indirectly because of a dependency.  All the tasks that use this value now have a dependency set on the `setGeeVersion` task so calling them even during an incremental run will trigger `setGeeVersion` to execute and update that property to its new value (if it changed) which should then trigger the other tasks to now run because that value changed.

To be sure I'll test the above senior to ensure it works as expected.

Reply:
Yes, I can see that's the code you've added, but that's not what I was describing.

When running a task is requested, such as, for example, `openGeeFusionRpm`, the build system needs to decide whether executing the operations defined for that task need to be run in the first place, or not.  That's typically done by checking whether any of the task's input files have changed since the task's output files were generated.  If inputs have not changed since the output was generated, the task won't run at all.  So, in this case, `openGeeFusionRpm` won't run.

There are more details about task execution.  I was describing a scenario for illustration, not to say that's the only problem that can result from this change.  I'm saying that there's a reason why some values are set during configuration, and there are consequences that may not be immediately apparent to consider.

Besides that, I think this adding more indirection for developers reading the code to follow, and adding more code to tasks.  And, I see little benefit resulting from these changes.

Reply:
I will re-do this and do another PR where the version information is generated during the configuration phase.  BTW I was playing with things and it appears that the package tasks always rebuild the rpm files.  Even when on files (input or output) have changed.  This might be a bug with the Netflix RPM plugin we are using...

Reply:
:laughing: There you go, one mess solved by another.

I have lost count of the work-arounds I've had to do for OS Package.

Thanks for verifying.  Then, I'd be OK with these changes.  It would be great if we have a cleaner version later.

Reply:
:laughing: There you go, one mess solved by another.

I have lost count of the work-arounds I've had to do for OS Package.

Thanks for verifying.  Then, I'd be OK with these changes.  It would be great if we have a cleaner version later.